modules/runner/src/integrationTest/java/org/apache/ignite/internal/sql/engine/ItFunctionsTest.java

```diff
@@ -285,16 +285,37 @@ public void testTypeOf() {
         assertThrowsWithCause(() -> sql("SELECT TYPEOF(SELECT 1, 2)"), CalciteContextException.class);
     }
 
+    /**
+     * Tests for {@code SUBSTRING(str, start[, length])} function.
+     */
+    @Test
+    public void testSubstring() {
+        assertQuery("SELECT SUBSTRING('1234567', 1, 3)").returns("123").check();
+        assertQuery("SELECT SUBSTRING('1234567', 2)").returns("234567").check();
+        assertQuery("SELECT SUBSTRING('1234567', -1)").returns("1234567").check();
+        assertQuery("SELECT SUBSTRING(1000, 1, 3)").returns("100").check();
+
+        assertQuery("SELECT SUBSTRING(NULL FROM 1 FOR 2)").returns(null).check();
+        assertQuery("SELECT SUBSTRING('text' FROM 1 FOR null)").returns(null).check();
+        assertQuery("SELECT SUBSTRING('test' FROM null FOR 2)").returns(null).check();
+
+        // uncomment after https://issues.apache.org/jira/browse/CALCITE-5708 was merged.
```

Comment:
We don't track Calcite tickets. 
Maybe there Ignite ticket to fix the issue or to bump Calcite dependency version?